sys/fs/nullfs/null_vnops.c

```diff
@@ -484,7 +484,7 @@ null_open(struct vop_open_args *ap)
 
 	vp = ap->a_vp;
 	ldvp = NULLVPTOLOWERVP(vp);
-	retval = null_bypass(&ap->a_gen);
+	retval = null_bypass((struct vop_generic_args *)ap);
```

Comment:
I'm not actually convinced that this better communicates the intent to be honest, and the pattern of returning a pointer to the subobject so you get type-checking from the compiler instead of a cast that can go stale is fairly common in the tree.

Reply:
Oh, if this is making the file more consistent, then yes, but maybe make the commit log say that. :)  Also, that change can definitely go upstream.

Reply:
5 casts and 3 &a_gen so it's a bit of a toss up, just need to pick one and go with it

Reply:
Agreed on being consistent.  Should be an easy change to upstream as well.

Reply:
I have corrected a commit message to mention that we want to be consistent here.

Should we only merge vnode_if.awk in this PR, and upstream null_vnops.c change to FreeBSD and not merge it to CheriBSD now?

Reply:
I would probably still drop the last paragraph from the commit log. :)  That said, fine to merge it now to CheriBSD and do an upstream review independently.  I suspect kib@ will have opinions on the upstream review.

Reply:
Done. Thanks!